demos/notebooks/demo_dendritic.ipynb

```diff
@@ -164,11 +201,27 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "#%% start a cluster for parallel processing (if a cluster already exists it will be closed and a new session will be opened)\n",
-    "if 'dview' in locals():\n",
+    "print(f\"You have {psutil.cpu_count()} CPUs available in your current environment\")\n",
+    "processors_to_use = psutil.cpu_count() - 4 #  None  # Set to number less than cpu_count - 1 if you have memory problems"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#start a cluster for parallel processing \n",
+    "# note if a cluster already exists it will be closed so a new session will be opened\n",
+    "if 'dview' in locals():  # locals contains list of current local variables\n",
```

Comment:
why is it unclear if dview will have a value?

Reply:
Jupyter mayhem. People jumping around Juypyter, and `dview` in particular you don't want accumulating. If people are working linearly, at this point it won't have a value yet, so cluster will get started. But it's pretty common to hop around.